### PR TITLE
Ticket25552 034 03

### PR DESCRIPTION
**Pull request by dgoulet-tor**

### CODE REVIEW

File: src/or/hs_descriptor.c [hunk 1]
```diff
@@ -2603,3 +2603,31 @@ hs_desc_lspec_to_trunnel(const hs_desc_link_specifier_t *spec)
...
+const char *
+hs_desc_get_start_of_sig(const char *encoded_desc)
+{
+  const char *start_of_sig;
+
+  tor_assert(encoded_desc);
+
+  /* We do look at newline + signature string so we make sure the siganture
+   * starts right after the encrypted MESSAGE. */
+  start_of_sig = tor_memstr(encoded_desc, strlen(encoded_desc),
```
[the remark this answers is not in the document]
Comment (dgoulet-tor) [reply src/or/hs_descriptor.c]: This is what this snippet of code does... It looks for `\n str_signature`.

File: src/or/hs_cache.c [hunk 2]
```diff
@@ -683,13 +683,6 @@ cache_store_as_client(hs_cache_client_descriptor_t *client_desc)
    * check if this descriptor is newer than the cached one */
   cache_entry = lookup_v3_desc_as_client(client_desc->key.pubkey);
   if (cache_entry != NULL) {
-    /* If we have an entry in our cache that has a revision counter greater
```
[the remark this answers is not in the document]
Comment (dgoulet-tor) [reply src/or/hs_cache.c]: Hmmm we aren't doing that within this patch. The goal is to make clients to not depend on revision counter anymore and just always replace the cached descriptor with the one downloaded.

Even though revision counter will still be used by services, at some point in time, they won't and client should be able to consider them obsolete.

File: src/or/hs_descriptor.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgoulet-tor) [reply src/or/hs_descriptor.c]: Oooook disregard this, I just noticed the extra `" "` at the end...

File: src/or/hs_cache.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (dgoulet-tor) [reply src/or/hs_cache.c]: See trac ticket for better explanation.

File: src/or/hs_descriptor.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgoulet-tor) [reply src/or/hs_descriptor.c]: Fixed in `23a4c09edb2c9d46`

**Comment by teor2345** [comment]
Obsoleted by #150

**Pull request closed without merging**